scripts/cangen/cangen/can_generator.py

```diff
@@ -220,39 +225,45 @@ def generate_code(bus: Bus, config: Config):
         "bus_name": bus.bus_name,
     }
 
-    logger.debug("Generating code for can messages")
-    _generate_from_jinja2_template(
-        CAN_MESSAGES_TEMPLATE_FILENAME,
-        os.path.join(config.output_dir, bus.bus_name.lower() + CAN_MESSAGES_FILE_NAME),
-        context,
-    )
+    logger.debug("Generating code for can messages and msg registry.")
 
-    logger.debug("Generating code for msg registry")
-    _generate_from_jinja2_template(
-        MSG_REGISTRY_TEMPLATE_FILENAME,
-        os.path.join(config.output_dir, bus.bus_name.lower() + MSG_REGISTRY_FILE_NAME),
-        context,
+    env = _create_jninja_environment(
+        PACKAGE_NAME, {"camel_to_snake": _camel_to_snake, "decimal_to_hex": hex}
     )
+    for template_file_name in TEMPLATE_FILE_NAMES:
+        _generate_jninja_template(
+            template_file_name,
+            _create_output_file_name(
+                config.output_dir, bus.bus_name, template_file_name
+            ),
+            context,
+            env,
+        )
 
     logger.info("Code generation complete")
 
 
-def _prepare_output_directory(output_dir):
+def _prepare_output_directory(output_dir: str):
     """Deletes previously generated files and creates a gitignore for the directory"""
     if os.path.exists(output_dir):
+        logger.info("Deleting previously generated code")
         shutil.rmtree(output_dir)
+
+    logger.info("Creating generated/can folder")
     os.makedirs(output_dir, exist_ok=True)
 
     gitignore_path = os.path.join(output_dir, ".gitignore")
     with open(gitignore_path, "w") as f:
         f.write("*")
 
 
-def generate_can_from_dbc(project_folder_name: str):
+def generate_can_for_project(project_folder_name: str):
+    """Generates C code for a given project.
+    Ensure the project folder contains a config.yaml file which specifies the relative path to its corresponding dbc file."""
     os.chdir(project_folder_name)
     config = Config.from_yaml("config.yaml")
 
     _prepare_output_directory(config.output_dir)
 
```

Comment:
now that the folder is ready, lets `chdir` into it here. There we don't have to worry about prepending every file path with `config.output_dir` anymore. This actually means that we don't need to pass `config` into `_generate_code()` anymore!